bubble_sort.rb

```diff
@@ -0,0 +1,38 @@
+def bubble_sort(array)
+  array_length = array.length
+  return array if array_length <= 1
+
+  swap = true
+  while swap
+    swap = false
+    (array_length - 1).times do |i|
+      if array[i] > array[i + 1]
+        array[i], array[i + 1] = array[i + 1], array[i]
+        swap = true
+      end
+    end
+  end
+  array
+end
+
+p bubble_sort([8, 6, 9, 1, 21])
+
+def bubble_sort_by(array)
+  array_length = array.length
+  return array if array_length <= 1
+
+  swap = true
+  while swap
+    swap = false
+    (array_length - 1).times do |i|
+      sorted_array = yield(array[i], array[i + 1])
+      if sorted_array.positive?
+        array[i], array[i + 1] = array[i + 1], array[i]
+        swap = true
+      end
+    end
+  end
+  array
+end
+
+p bubble_sort_by(%w[hello nope j]) { |left, right| left.length - right.length }
```

Comment:
In my own opinion, I think you should remove lines 18 and 38. I believe you used these lines for testing purposes, and that should be it. They should be removed because the algorithm is not limited to this set of data alone. Let the users decide what data to test your algorithm with.